xcdat/axis.py

```diff
@@ -365,37 +504,10 @@ def _align_lon_bounds_to_360(
     # Update the lower-most min and upper-most max bounds to [0, 360].
     # Result: [[0, 1], [1, 90], [90, 180], [180, 359], [359, 360]]
     bounds[p_meridian_index, 0], bounds[-1, 1] = (0.0, 360.0)
-    return bounds
-
-
-def _get_prime_meridian_index(lon_bounds: xr.DataArray) -> Optional[np.ndarray]:
-    """Gets the index of the prime meridian cell in the longitude bounds.
-
-    A prime meridian cell can exist when converting the axis orientation
-    from [-180, 180) to [0, 360).
-
-    Parameters
-    ----------
-    lon_bounds : xr.DataArray
-        The longitude bounds.
-
-    Returns
-    -------
-    Optional[np.ndarray]
-        An array with a single elementing representing the index of the prime
-        meridian index if it exists. Otherwise, None if the cell does not exist.
 
-    Raises
-    ------
-    ValueError
-        If more than one grid cell spans the prime meridian.
-    """
-    p_meridian_index = np.where(lon_bounds[:, 1] - lon_bounds[:, 0] < 0)[0]
+    # Update last value of each coordinate to 360.
+    for key, coord in bounds.coords.items():
+        coord.values[-1] = 360
+        bounds[key] = coord
```

Comment:
This block raises an index error when I call `spatial.average` (file listed below):

> File ~/code/xcdat/robust_tests/../xcdat/axis.py:510, in _align_lon_bounds_to_360(bounds, p_meridian_index)
>     508 # Update last value of each coordinate to 360.
>     509 for key, coord in bounds.coords.items():
> --> 510     coord.values[-1] = 360
>     511     bounds[key] = coord
>     513 return bounds
> 
> IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed

`/p/css03/esgf_publish/CMIP6/CMIP/CSIRO-ARCCSS/ACCESS-CM2/historical/r1i1p1f1/Amon/tas/gn/v20191108/tas_Amon_ACCESS-CM2_historical_r1i1p1f1_gn_185001-201412.nc` yields an error when calling `spatial.average` (once you get past the issue with the singleton bounds on the `height` domain).

Reply:
I think I added this code because I was running into the issue where swapping coordinates and bounds from (0, 360) to (0, 360) results in the last value being 0 rather than 360 (this [PR comment](https://github.com/xCDAT/xcdat/pull/343#discussion_r980352049)). As a result, the last value needed to be updated back to 360. 

**This block of code should be removed since `_swap_lon_axis()` handles this now**:
https://github.com/xCDAT/xcdat/blob/75e426a6243adffd503fc539229398ebc43c070e/xcdat/axis.py#L311-L327

Hopefully this makes sense. Thanks for the catch!

Reply:
> 
> This block raises an index error when I call spatial.average (file listed below):
> 
> File ~/code/xcdat/robust_tests/../xcdat/axis.py:510, in _align_lon_bounds_to_360(bounds, p_meridian_index)
> 508 # Update last value of each coordinate to 360.
> 509 for key, coord in bounds.coords.items():
> --> 510 coord.values[-1] = 360
> 511 bounds[key] = coord
> 513 return bounds
> 
> IndexError: too many indices for array: array is 0-dimensional, but 1 were indexed

I actually never run into your error with this code block on the latest commit on this branch. 
Did you perform some operation before calling `ds.spatial.average`?

Reply:
I may have caused this while tinkering with your PR (to address the singleton coord issue). I just switched to your latest code and I am not having this issue now (even though it still has this block of code).